sql/atxs/atxs_test.go

```diff
@@ -1201,3 +1201,56 @@ func Test_AtxWithPrevious(t *testing.T) {
 		require.Equal(t, atx2.ID(), id)
 	})
 }
+
+func Test_FindDoublePublish(t *testing.T) {
+	t.Parallel()
+	sig, err := signing.NewEdSigner()
+	require.NoError(t, err)
+	t.Run("no atxs", func(t *testing.T) {
+		t.Parallel()
+		db := sql.InMemory()
+		_, err := atxs.FindDoublePublish(db, types.RandomNodeID(), 0)
+		require.ErrorIs(t, err, sql.ErrNotFound)
+	})
+
+	t.Run("no double publish", func(t *testing.T) {
+		t.Parallel()
+		db := sql.InMemory()
+
+		// one atx
+		atx0, blob := newAtx(t, sig, withPublishEpoch(1))
```

Comment:
`newAtx` returns a `wire.ActivationTxV1` as blob. I think `newAtx` doesn't need to return any wire type for the tests here, if a blob is needed for a test it should generally be possible to just use `types.RandomBytes(20)` since the blob should not be needed to be decoded in any test in this package.

Wdyt?

Reply:
I wouldn't say I like returning a fake value when a real one is so easy to use. It's a footgun waiting to shoot somebody in the future. However, the blob could be obtained by calling `atx0.Blob()`, so we can only return the ATX.

I will refactor it to return only ATX in a separate PR.

Reply:
The problem I see with returning a real value is that for this we probably will need a dependency on `activation` and/or `activation/wire` but `sql` packages shouldn't depend on those (cyclic dependencies). Using `xxx_test` packages for tests is a way around this, but since sql packages can't even decode the bytes by themselves I don't see an issue with faking the bytes that are stored / returned in tests.